packages/core/core/src/BundleGraph.js

```diff
@@ -629,30 +689,36 @@ export default class BundleGraph {
   }
 
   isAssetReferencedByDependant(bundle: Bundle, asset: Asset): boolean {
-    let assetNode = nullthrows(this._graph.getNode(asset.id));
+    let assetNodeId = nullthrows(this._graph.getNodeIdByContentKey(asset.id));
     let dependencies = this._graph
-      .getNodesConnectedTo(assetNode)
+      .getNodeIdsConnectedTo(assetNodeId)
+      .map(id => nullthrows(this._graph.getNode(id)))
       .filter(node => node.type === 'dependency')
       .map(node => {
         invariant(node.type === 'dependency');
         return node.value;
       });
 
     // Collect bundles that depend on this asset being available to reference
-    // asynchonously. If any of them appear in our traversal, this asset is
+    // asynchronously. If any of them appear in our traversal, this asset is
     // referenced.
     let asyncInternalReferencingBundles = new Set(
       this._graph
-        .getNodesConnectedTo(assetNode, 'references')
+        .getNodeIdsConnectedTo(assetNodeId, 'references')
+        .map(id => nullthrows(this._graph.getNode(id)))
         .filter(node => node.type === 'dependency')
         .map(node => {
           invariant(node.type === 'dependency');
           return node;
         })
         .flatMap(dependencyNode =>
           this._graph
-            .getNodesConnectedTo(dependencyNode, 'internal_async')
-            .map(node => {
+            .getNodeIdsConnectedTo(
+              this._graph.getNodeIdByContentKey(dependencyNode.id),
+              'internal_async',
+            )
+            .map(id => {
+              let node = nullthrows(this._graph.getNode(id));
```

Comment:
There are a few places where `getNodesConnectedTo` is now replaced with `getNodeIdsConnectedTo().map(id => getNode(id))`. Maybe don't remove `getNodesConnectedTo`?